File: jzy-cart-interface/src/main/java/cn/jzy/cart/service/CartService.java
package cn.jzy.cart.service;

import cn.jzy.pojogroup.Cart;

import java.util.List;

/**
 * @Auther: 林俊豪
 * @Date: 2019/6/13
 * 购物车服务接口
 */

public interface CartService {

    /**
     *  添加商品到购物车
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num);

    /**
     *  从redis中提取购物车
     * @return
     */
    List<Cart> findCartListFromRedis(String username);

    /**
     *  将购物车列表存入Redis
     * @param username
     * @param cartList
     */
    void saveCartListToRedis(String username, List<Cart> cartList);


    /**
     *  合并购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);
}

File: jzy-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService', function ($http) {

    // 购物车列表
    this.findCartList = function () {
        return $http.get('cart/findCartList.do');
    };

    // 添加商品到购物车
    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('/cart/addGoodsToCartList.do?itemId=' + itemId + "&num=" + num);
    };

    // 求和
    this.sum = function (cartList) {
        var totalValue = {totalNum : 0, totalMoney : 0};

        for (var i = 0; i < cartList.length; i ++){
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j ++){
                var orderItem = cart.orderItemList[j];
                totalValue.totalNum += orderItem.num; // 累加数量
                totalValue.totalMoney += orderItem.totalFee; // 累加金额
            }
        }
        return totalValue;
    }
});